ext/json/json.c

```diff
@@ -270,6 +292,66 @@ PHP_FUNCTION(json_decode)
 }
 /* }}} */
 
+/* {{{ Validates if an string contains a valid json */
+PHP_FUNCTION(json_validate)
+{
+	char *str;
+	size_t str_len;
+	zend_long depth = PHP_JSON_PARSER_DEFAULT_DEPTH;
+	zend_long options = 0;
+
+	ZEND_PARSE_PARAMETERS_START(1, 3)
+		Z_PARAM_STRING(str, str_len)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_LONG(depth)
+		Z_PARAM_LONG(options)
+	ZEND_PARSE_PARAMETERS_END();
+
+
+	if (options != 0) {
+		zend_long tmp_options = options;
+		tmp_options ^= (PHP_JSON_INVALID_UTF8_IGNORE | PHP_JSON_THROW_ON_ERROR);
+
+		if (tmp_options != 0) {
+			if (!((tmp_options == PHP_JSON_INVALID_UTF8_IGNORE) || (tmp_options == PHP_JSON_THROW_ON_ERROR))) {
+				zend_argument_value_error(3, "must be a valid flag (JSON_THROW_ON_ERROR, JSON_INVALID_UTF8_IGNORE)");
+				RETURN_THROWS();
+			}
+		}
+	}
+
+	if (!str_len) {
+		if (!(options & PHP_JSON_THROW_ON_ERROR)) {
+			JSON_G(error_code) = PHP_JSON_ERROR_SYNTAX;
+		} else {
+			zend_throw_exception(php_json_exception_ce, php_json_get_error_msg(PHP_JSON_ERROR_SYNTAX), PHP_JSON_ERROR_SYNTAX);
+		}
+
+		RETURN_FALSE;
+	}
+
+	if (!(options & PHP_JSON_THROW_ON_ERROR)) {
+		JSON_G(error_code) = PHP_JSON_ERROR_NONE;
```

Comment:
You're setting the `error_code` here, but wouldn't it make more sense to *return* this value from the function? That would mean that instead of returning a boolean, it would have to be an integer, with `0` likely having to mean `no error`, but that is not something that is unheard of.

Reply:
I have to say that returning a boolean was my personal preferred approach. If you think we should do it with the **int** way, let me know.

Reply:
I have being thinking, and I really prefer returning bool, and being able to call json_last_error() to find out what happened

Reply:
I wrote this down in the RFC as an open question